mongo_test.go

```diff
@@ -170,26 +204,36 @@ func TestDelete(t *testing.T) {
 	}
 
 	// Can't delete a non existing item
-	err = h.Delete(context.Background(), item)
-	assert.Equal(t, resource.ErrNotFound, err)
+	err := h.Delete(context.Background(), item)
+	if expect := resource.ErrNotFound; err != expect {
+		t.Errorf("Unexpected error:\nwant: %s\ngot: %s", expect, err)
+	}
 
 	err = h.Insert(context.Background(), []*resource.Item{item})
-	assert.NoError(t, err)
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
 	err = h.Delete(context.Background(), item)
-	assert.NoError(t, err)
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
 
 	// Update refused if original item's etag doesn't match stored one
 	err = h.Insert(context.Background(), []*resource.Item{item})
-	assert.NoError(t, err)
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
 	item.ETag = "etag2"
 	err = h.Delete(context.Background(), item)
-	assert.Equal(t, resource.ErrConflict, err)
+	if expect := resource.ErrConflict; err != expect {
```

Comment:
The idiomatic construct for this is:

```go
if got, want := err, resource.ErrConflict; got != want {
    t.Errorf("Handler.Delete err = : %v, want: %v", got, want)
}
```

Reply:
I'll see what I can do about that.

Reply:
Updated. Looked at tests in the the `encoding/json` for inspiration. I noticed that they report errors of the got/want format with a space prefix:

```go
t.Errorf(" got: %v want: %v", result, expect)
```

Or, for longer results:
```go
t.Errorf(" got: %v\nwant: %v\n", result, expect)
```

I haven't including the space prefix, but can do so if you prefer that. I have however added a newline prefix when appropriate, as that looks better in the test output to me:
```go
t.Errorf("\ngot: %v\nwant: %v\n", result, expect)
```

Giving output ala:

```sh
Running tool: /usr/local/bin/go test -timeout 30s github.com/rs/rest-layer-mongo -run ^(TestUpdate)$

--- FAIL: TestUpdate (0.11s)
    --- FAIL: TestUpdate/when_updating_an_existing_item (0.05s)
        --- FAIL: TestUpdate/when_updating_an_existing_item/should_have_updated_the_item (0.00s)
            /Users/smyrman/Code/rest-layer-mongo/mongo_test.go:178: 
                got: map[_updated:2019-01-03 13:32:34.858 +0100 CET _id:1234 foo:baz _etag:etag2]
                want: map[_id:1234 _etag:etag2-printerrrors _updated:2019-01-03 13:32:34.858 +0100 CET foo:baz]
    --- FAIL: TestUpdate/when_updating_an_item_that_doesn't_have_an_E-Tag_set_in_the_DB (0.05s)
        --- FAIL: TestUpdate/when_updating_an_item_that_doesn't_have_an_E-Tag_set_in_the_DB/should_have_updated_the_item (0.00s)
            /Users/smyrman/Code/rest-layer-mongo/mongo_test.go:225: 
                got: map[_id:1234 foo:baz _etag:etag2 _updated:2019-01-03 13:32:34.858 +0100 CET]
                want: map[_id:1234 _etag:etag2-printerrrors _updated:2019-01-03 13:32:34.858 +0100 CET foo:baz]
FAIL
FAIL	github.com/rs/rest-layer-mongo	0.128s
Error: Tests failed.
```